src/screens/dialogs/move-file-dialog.tsx

```diff
@@ -0,0 +1,154 @@
+import { useContext, useEffect, useState } from "react";
+import { moveFileService } from "../../services/files/move-file.service";
+import { listFilesService } from "../../services/files/list-file.service";
+import { useSearchParams } from "react-router-dom";
+import { Dialog } from "../../components/Dialog";
+import { File } from "../../types/entities";
+import { FolderPlus } from "lucide-react";
+import {
+  FilesDialogsContext,
+  AVAILABLE_DIALOGS,
+  AuthContext,
+  FilesContext
+} from "../../context/index";
+import toast from "react-hot-toast";
+
+export const MoveFileDialog = () => {
+  const { removeFile } = useContext(FilesContext);
+  const [searchParams, _setSearchParams] = useSearchParams();
+  const { dialogsVisibilityState, closeDialog, selectedFile } =
+    useContext(FilesDialogsContext);
+  const Open = dialogsVisibilityState[AVAILABLE_DIALOGS.MOVE_FILE];
+  const { session } = useContext(AuthContext);
+  const [folders, setFolders] = useState<File[]>([]);
+  const [moving, setMoving] = useState(false);
+  const [selectedFolder, setSelectedFolder] = useState<File | null>(null);
+  const directory = searchParams.get("directory");
+
+  useEffect(() => {
+    if (Open && selectedFile) {
+      const fetchFolders = async () => {
+        const { success, ...res } = await listFilesService({
+          token: session?.token as string,
+          directory: selectedFolder ? selectedFolder.uuid : directory
+        });
+
+        if (!success) {
+          toast.error(res.msg);
+        } else {
+          const folderList = res.files.filter(
+            (file) => !file.isFile && file.uuid !== selectedFile?.uuid
+          );
+          setFolders(folderList);
+        }
+      };
+
+      fetchFolders();
+    }
+  }, [directory]);
```

Comment:
It seems that you're not using the directory parameter to navigate between the directories so don't use it here. Just use the `selectedFolder`.